contracts/strategies/examples/sqf-superfluid/SQFSuperfluid.sol

```diff
@@ -0,0 +1,451 @@
+
+        Recipient storage recipient = _recipients[recipientId];
+
+        _updateMemberUnits(recipientId, recipient.recipientAddress, uint128(recipientTotalUnits));
+
+        uint256 currentFlowRate = recipientFlowRate[recipientId];
+
+        recipientFlowRate[recipientId] = currentFlowRate + _newFlowRate - _previousFlowRate;
+
+        emit TotalUnitsUpdated(recipientId, recipientTotalUnits);
+    }
+
+    /// @notice Close the stream
+    function closeStream() external onlyPoolManager(msg.sender) {
+        poolSuperToken.distributeFlow(address(this), gdaPool, 0, "0x");
+        address(poolSuperToken).transferAmount(msg.sender, poolSuperToken.balanceOf(address(this)));
+    }
+
+    /// @notice Checks if the allocator is valid
+    /// @param _allocator The allocator address
+    /// @return 'true' if the allocator is valid, otherwise 'false'
+    function isValidAllocator(address _allocator) external view returns (bool) {
+        return _isValidAllocator(_allocator);
+    }
+
+    /// ====================================
+    /// ============ Internal ==============
+    /// ====================================
+
+    /// @dev prevent the pool token from being withdrawn
+    /// @param _token The token address
+    /// @param _amount The amount to withdraw
+    /// @param _recipient The address to withdraw to
+    function _beforeWithdraw(address _token, uint256 _amount, address _recipient) internal override {
+        if (_token == address(poolSuperToken)) {
+            revert INVALID();
+        }
+    }
+
+    /// @notice Checks if the allocator is valid
+    /// @param _allocator The allocator address
+    /// @return 'true' if the allocator is valid, otherwise 'false'
+    function _isValidAllocator(address _allocator) internal view override returns (bool) {
+        // Check if the allocator has a minimum passport score
+        uint256 allocatorScore = passportDecoder.getScore(_allocator);
+        if (allocatorScore < minPassportScore) {
+            // If the allocator does not have the required score, then it is not valid. No need to check for NFTs
+            return false;
+        }
+
+        // Check if the allocator has the required NFTs
+        for (uint256 i; i < erc721s.length; i++) {
+            _checkOnlyWithNFT(erc721s[i], _allocator);
+        }
+        // If didnt revert, then the allocator has the required NFT (and we assume it also has the required score)
+        return true;
+    }
+
+    /// @dev If the recipient is accepted, create a super app for the recipient
+    /// @param _newStatus The new status
+    /// @param _oldStatus The old status
+    /// @param _recipientIndex The index of the recipient
+    /// @return _reviewedStatus The reviewed status
+    function _reviewRecipientStatus(Status _newStatus, Status _oldStatus, uint256 _recipientIndex)
+        internal
+        override
+        nonReentrant
+        returns (Status _reviewedStatus)
+    {
+        if (_newStatus == Status.Accepted) {
```

Comment:
`Status.Canceled` is allowed but not handled. I suggest something like the following: (1) move the `cancelRecipients()` logic to `_reviewRecipientStatus()`, (2) override `_validateReviewRecipients()` to include both onlyActiveAllocation and onlyActiveRegistration, (3) revert status updates different than `Status.Canceled` during allocations.

Reply:
lets leave it as is for now, as it follows their existing strategy, and we can suggest the improvement

Reply:
I think it's a bug, not an improvement. If they "cancel" an accepted recipient calling `reviewRecipients()`, the recipient won't get actually canceled and they won't be able to cancel it with `cancelRecipients()` afterwards. If not fixed, at least we should place a big warning somewhere.

Reply:
I believe this is expected to happen since that's how they have been using the strategy in the past